packages/vitest/src/utils/graph.ts

```diff
@@ -29,7 +29,18 @@ export async function getModuleGraph(ctx: Vitest, id: string): Promise<ModuleGra
     graph[id] = (await Promise.all(mods.map(m => get(m, seen)))).filter(Boolean) as string[]
     return id
   }
-  await get(ctx.server.moduleGraph.getModuleById(id))
+
+  const workspaceProject = ctx.projects.find(project => project.getName() === workspaceName)
+  const workspaceModules = workspaceProject?.ctx.getModuleProjects(id)
+  const selectedModule = workspaceModules?.find(module => module.getName() === workspaceName)
```

Comment:
You do the same thing you did on #L33. Why?

You need to find the project by name and fallback to `ctx.getCoreWorkspaceProject().server || ctx.server`

Reply:
@sheremet-va  `getCoreWorkspaceProject()` is coming as `null` for `test/workspaces` folder.
![Screenshot 2023-08-08 213513](https://github.com/vitest-dev/vitest/assets/61521805/0f736194-de51-4cbe-b4a6-dce74eb7cc3f)

Upon investigating, I have found that the conditons that executes the `createCoreWorkspace` function in `packages\vitest\src\node\core.ts` aren't satisfying

![Screenshot 2023-08-08 213414](https://github.com/vitest-dev/vitest/assets/61521805/7a436d84-1974-4613-9116-5115b25fc532)

> You do the same thing you did on #L33. Why?

This might be hacky, but I have found only this way where we get the `workspaceMod`

Reply:
> `getCoreWorkspaceProject()` is coming as `null` for `test/workspaces` folder.

Yes, it will be null if there is no core project. This is why I added `|| ctx.server`. Core project is not created if main config file is not used by a workspace (so, when there is a global config and project configs and workspace specifies only project configs without manually using core project)

> This might be hacky, but I have found only this way where we get the `workspaceMod`

Then you must be doing something wrong?

Reply:
Workspace project is just a project with the same name. If there is no such projects, just fallback to context which is always defined